packages/sdk-communication-layer/src/services/SocketService/ConnectionManager/connectToChannel.ts

```diff
@@ -61,23 +63,67 @@ export function connectToChannel({
   instance.state.isOriginator = isOriginator;
   instance.state.channelId = channelId;
   setupChannelListeners(instance, channelId);
-  instance.state.socket?.emit(
-    EventType.JOIN_CHANNEL,
-    {
-      channelId,
-      context: `${instance.state.context}_connectToChannel`,
-      clientType: isOriginator ? 'dapp' : 'wallet',
-    },
-    (
-      error: string | null,
-      result?: { ready: boolean; persistence: boolean },
-    ) => {
-      if (error === 'error_terminated') {
-        instance.emit(EventType.TERMINATE);
-      } else if (typeof result === 'object' && result.persistence) {
-        // Inform that this channel supports full session persistence
-        instance.emit(EventType.CHANNEL_PERSISTENCE);
-      }
-    },
-  );
+
+  if (!isOriginator && authorized) {
+    instance.state.keyExchange?.setKeysExchanged(true);
+    instance.remote.state.ready = true;
+    instance.remote.state.authorized = true;
+  }
+
+  return new Promise((resolve) => {
```

Comment:
I think we should refactor this section to improve readability and maintainability. Specifically, let's consider extracting the callback logic from the return Promise into a separate, named function. This will make the code more readable, easier to test, and more maintainable in the long term.

Additionally, using async/await directly would simplify the Promise handling, making the flow of the code clearer.

WDYT? @abretonc7s